service/model/project_membership.go

```diff
@@ -115,3 +123,48 @@ func (i ProjectInvitationStatus) Validate() error {
 
 	return errProjectInvitationStatusInvalid
 }
+
+func (r ProjectRole) IsRoleAtLeast(role ProjectRole) bool {
+	return projectRolePriority[r] <= projectRolePriority[role]
+}
+
+type ProjectMember struct {
+	UserID      uuid.UUID
+	ProjectID   uuid.UUID
+	ProjectRole ProjectRole
+	UserData    UserData
+	CreatedAt   time.Time
+	UpdatedAt   time.Time
+}
+
+type UserData struct {
+	Name  string
+	Email string
+}
+
+func NewProjectMember(userID, projectID uuid.UUID, role ProjectRole) ProjectMember {
+	now := time.Now()
```

Comment:
And when creating a new member, i don't need to have whole user object, user id is enough.

If I want to create member also with user data, I can do `NewProjectMember(...).AddUserData(..)`